dbms/src/Storages/KVStore/MultiRaft/RegionSerde.cpp

```diff
@@ -0,0 +1,276 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/FailPoint.h>
+#include <Storages/KVStore/Region.h>
+#include <Storages/KVStore/Utils/SerializationHelper.h>
+
+#include <memory>
+#include <utility>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int UNKNOWN_FORMAT_VERSION;
+} // namespace ErrorCodes
+
+namespace FailPoints
+{
+extern const char force_region_persist_version[];
+extern const char force_region_read_version[];
+extern const char force_region_persist_extension_field[];
+extern const char force_region_read_extension_field[];
+} // namespace FailPoints
+
+enum class RegionPersistVersion
+{
+    V1 = 1,
+    V2, // For eager gc
+};
+
+namespace RegionPersistFormat
+{
+static constexpr UInt32 HAS_EAGER_TRUNCATE_INDEX = 0x01;
+// The upper bits are used to store length of extensions. DO NOT USE!
+} // namespace RegionPersistFormat
+
+using MaybeRegionPersistExtension = UInt32;
+enum class RegionPersistExtension : MaybeRegionPersistExtension
+{
+    Reserved1 = 1,
+    ReservedForTest = 2,
+    // It should always be equal to the maximum supported type + 1
+    MaxKnownFlag = 3,
```

Comment:
I want a manual check here. Consider it we deprecated some enums like
```
enum class {
    A = 1,
    // B = 2, // deprecated since V100
    C = 3,
}
```
It could result in silent error.